src/main/c/internal/ZlibContext.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <zlib.h>
+
+#include <vector>
+#include <stdexcept>
+
+
+class ZlibContext {
+public:
```

Comment:
Yes I agree, this should hopefully never be copied.

Also, it just occurred to me that this class is living in a global namespace.. You could wrap that in a seasocks namespace if you like.